File: android/employer/app/src/main/java/com/workforcex/employer/ui/employer/VerifyWorkerActivity.java
package com.workforcex.employer.ui.employer;

import android.os.Bundle;
import android.view.View;
import android.widget.Toast;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import com.workforcex.employer.api.RetrofitClient;
import com.workforcex.employer.api.Verification;
import com.workforcex.employer.databinding.ActivityVerifyWorkerBinding;
import com.workforcex.employer.utils.TokenManager;
import java.util.List;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class VerifyWorkerActivity extends AppCompatActivity {

    private ActivityVerifyWorkerBinding binding;
    private TokenManager tokenManager;
    private String workerId;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityVerifyWorkerBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());
        tokenManager = new TokenManager(this);

        workerId = getIntent().getStringExtra("workerId");
        String workerName = getIntent().getStringExtra("workerName");
        setTitle("Verify " + workerName);

        binding.rvDocuments.setLayoutManager(new LinearLayoutManager(this));
        loadWorkerDocuments();
    }

    private void loadWorkerDocuments() {
        binding.progressBar.setVisibility(View.VISIBLE);
        // This endpoint doesn't exist yet, so we'll need to add it
        RetrofitClient.get().getWorkerDocuments(tokenManager.getBearerToken(), workerId)
                .enqueue(new Callback<List<Verification>>() {
                    @Override
                    public void onResponse(Call<List<Verification>> call, Response<List<Verification>> response) {
                        binding.progressBar.setVisibility(View.GONE);
                        if (response.isSuccessful() && response.body() != null) {
                            binding.rvDocuments.setAdapter(new DocumentAdapter(response.body(), VerifyWorkerActivity.this::updateVerificationStatus));
                        } else {
                            Toast.makeText(VerifyWorkerActivity.this, "Failed to load documents", Toast.LENGTH_SHORT).show();
                        }
                    }

                    @Override
                    public void onFailure(Call<List<Verification>> call, Throwable t) {
                        binding.progressBar.setVisibility(View.GONE);
                        Toast.makeText(VerifyWorkerActivity.this, "Network error", Toast.LENGTH_SHORT).show();
                    }
                });
    }

    private void updateVerificationStatus(String verificationId, String status, String comments) {
        RetrofitClient.get().updateEmployerVerificationStatus(tokenManager.getBearerToken(), verificationId, status, comments)
                .enqueue(new Callback<Void>() {
                    @Override
                    public void onResponse(Call<Void> call, Response<Void> response) {
                        if (response.isSuccessful()) {
                            Toast.makeText(VerifyWorkerActivity.this, "Verification status updated", Toast.LENGTH_SHORT).show();
                            loadWorkerDocuments(); // Refresh the list
                        } else {
                            Toast.makeText(VerifyWorkerActivity.this, "Failed to update status", Toast.LENGTH_SHORT).show();
                        }
                    }

                    @Override
                    public void onFailure(Call<Void> call, Throwable t) {
                        Toast.makeText(VerifyWorkerActivity.this, "Network error", Toast.LENGTH_SHORT).show();
                    }
                });
    }
}

File: android/employer/app/src/main/java/com/workforcex/employer/ui/employer/VerifyWorkerActivity.kt
package com.workforcex.employer.ui.employer

import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import com.workforcex.employer.databinding.ActivityVerifyWorkerBinding
import com.workforcex.employer.utils.TokenManager
import com.workforcex.shared_employer.RetrofitClient
import com.workforcex.shared_employer.models.Document
import com.workforcex.shared_employer.models.Verification
import com.workforcex.shared_employer.models.VerificationUpdateBody
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class VerifyWorkerActivity : AppCompatActivity() {

    private lateinit var binding: ActivityVerifyWorkerBinding
    private lateinit var tokenManager: TokenManager
    private lateinit var workerId: String

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityVerifyWorkerBinding.inflate(layoutInflater)
        setContentView(binding.root)
        tokenManager = TokenManager(this)

        workerId = intent.getStringExtra("workerId") ?: ""
        val workerName = intent.getStringExtra("workerName")
        title = "Verify $workerName"

        binding.rvDocuments.layoutManager = LinearLayoutManager(this)
        binding.rvUploadedFiles.layoutManager = LinearLayoutManager(this)
        loadWorkerDocuments()
        loadWorkerUploadedFiles()
    }

    private fun loadWorkerDocuments() {
        binding.progressBar.visibility = View.VISIBLE
        RetrofitClient.get().getWorkerDocuments(tokenManager.getBearerToken(), workerId)
            .enqueue(object : Callback<List<Verification>> {
                override fun onResponse(
                    call: Call<List<Verification>>,
                    response: Response<List<Verification>>
                ) {
                    binding.progressBar.visibility = View.GONE
                    if (response.isSuccessful && response.body() != null) {
                        binding.rvDocuments.adapter =
                            DocumentAdapter(response.body()!!, this@VerifyWorkerActivity::updateVerificationStatus)
                    } else {
                        Toast.makeText(this@VerifyWorkerActivity, "Failed to load documents", Toast.LENGTH_SHORT).show()
                    }
                }

                override fun onFailure(call: Call<List<Verification>>, t: Throwable) {
                    binding.progressBar.visibility = View.GONE
                    Toast.makeText(this@VerifyWorkerActivity, "Network error", Toast.LENGTH_SHORT).show()
                }
            })
    }

    private fun loadWorkerUploadedFiles() {
        RetrofitClient.get().getWorkerDocumentFiles(tokenManager.getBearerToken(), workerId)
            .enqueue(object : Callback<List<Document>> {
                override fun onResponse(
                    call: Call<List<Document>>,
                    response: Response<List<Document>>
                ) {
                    if (response.isSuccessful && response.body() != null) {
                        binding.rvUploadedFiles.adapter = WorkerFileAdapter(response.body()!!) { doc ->
                            openDocument(doc)
                        }
                    }
                }

                override fun onFailure(call: Call<List<Document>>, t: Throwable) {
                    // Non-critical: verification status list above still loads fine.
                }
            })
    }

    private fun openDocument(document: Document) {
        val fullUrl = if (document.fileUrl.startsWith("http")) {
            document.fileUrl
        } else {
            RetrofitClient.baseUrl().trimEnd('/') + document.fileUrl
        }
        try {
            startActivity(Intent(Intent.ACTION_VIEW, Uri.parse(fullUrl)))
        } catch (e: Exception) {
            Toast.makeText(this, "Unable to open document", Toast.LENGTH_SHORT).show()
        }
    }

    private fun updateVerificationStatus(verificationId: String, status: String, comments: String) {
        RetrofitClient.get().updateEmployerVerificationStatus(tokenManager.getBearerToken(), verificationId, status, VerificationUpdateBody(comments))
            .enqueue(object : Callback<Void> {
                override fun onResponse(call: Call<Void>, response: Response<Void>) {
                    if (response.isSuccessful) {
                        Toast.makeText(this@VerifyWorkerActivity, "Verification status updated", Toast.LENGTH_SHORT).show()
                        loadWorkerDocuments() // Refresh the list
                    } else {
                        Toast.makeText(this@VerifyWorkerActivity, "Failed to update status", Toast.LENGTH_SHORT).show()
                    }
                }

                override fun onFailure(call: Call<Void>, t: Throwable) {
                    Toast.makeText(this@VerifyWorkerActivity, "Network error", Toast.LENGTH_SHORT).show()
                }
            })
    }
}
